### PR TITLE
fix:  with empty solicitor's references, claim submit crashes (CMC-933)

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-933](t/jira/browse/CMC-933)

### Change description ###
This PR fixes the issue when we submit claim with empty solicitor's references.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java [hunk 1]
```diff
@@ -76,8 +79,12 @@ public SealedClaimForm getTemplateData(CaseData caseData) {
             .referenceNumber(caseData.getLegacyCaseReference())
             .issueDate(caseData.getClaimIssuedDate())
             .submittedOn(caseData.getClaimSubmittedDateTime().toLocalDate())
-            .claimantExternalReference(caseData.getSolicitorReferences().getApplicantSolicitor1Reference())
-            .defendantExternalReference(caseData.getSolicitorReferences().getRespondentSolicitor1Reference())
+            .claimantExternalReference(solicitorReferences
+                                           .map(SolicitorReferences::getApplicantSolicitor1Reference)
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:83]: should this not be something like "not provided"

File: src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java [hunk 2]
```diff
@@ -65,6 +67,7 @@ private String getFileName(CaseData caseData) {
 
     @Override
     public SealedClaimForm getTemplateData(CaseData caseData) {
+        Optional<SolicitorReferences> solicitorReferences = Optional.ofNullable(caseData.getSolicitorReferences());
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:70]: is this an issue in confirmation of service too?

File: src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java (hunk 1 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:83]: Nope that's fine.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:83]: "not provided" was for COS requirements and not for sealed claim.

File: src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:70]: Good catch. yes it is broken there as well. fixing

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/sealedclaim/SealedClaimFormGenerator.java:70]: fixed

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**